src/query/functions-v2/src/scalars/variant.rs

```diff
@@ -43,4 +75,756 @@ pub fn register(registry: &mut FunctionRegistry) {
+                        Some(keys) => {
+                            bitmap.push(true);
+                            builder.put_slice(&keys);
+                        }
+                        None => {
+                            bitmap.push(false);
+                        }
+                    }
+                    builder.commit_row();
+                }
+                let col = NullableColumn {
+                    validity: bitmap.into(),
+                    column: builder.build(),
+                };
+                Ok(Value::Column(col))
+            }
+            ValueRef::Scalar(v) => {
+                let val = match object_keys(v) {
+                    Some(keys) => Value::Scalar(Some(keys)),
+                    None => Value::Scalar(None),
+                };
+                Ok(val)
+            }
+        },
+    );
+
+    registry.register_1_arg_core::<NullableType<VariantType>, NullableType<VariantType>, _, _>(
+        "object_keys",
+        FunctionProperty::default(),
+        |_| None,
+        |arg, _| match &arg {
+            ValueRef::Column(NullableColumn { validity, column }) => {
+                let size = column.len();
+                let mut bitmap = MutableBitmap::with_capacity(size);
+                let mut builder = StringColumnBuilder::with_capacity(size, 0);
+                for idx in 0..size {
+                    let b = unsafe { validity.get_bit_unchecked(idx) };
+                    if !b {
+                        bitmap.push(false);
+                        builder.commit_row();
+                        continue;
+                    }
+                    let v = unsafe { column.index_unchecked(idx) };
+                    if v.is_empty() {
+                        bitmap.push(false);
+                        builder.commit_row();
+                        continue;
+                    }
+                    match object_keys(v) {
+                        Some(keys) => {
+                            bitmap.push(true);
+                            builder.put_slice(&keys);
+                        }
+                        None => {
+                            bitmap.push(false);
+                        }
+                    }
+                    builder.commit_row();
+                }
+                let col = NullableColumn {
+                    validity: bitmap.into(),
+                    column: builder.build(),
+                };
+                Ok(Value::Column(col))
+            }
+            ValueRef::Scalar(None) => Ok(Value::Scalar(None)),
+            ValueRef::Scalar(Some(v)) => {
+                let val = match object_keys(v) {
+                    Some(keys) => Value::Scalar(Some(keys)),
+                    None => Value::Scalar(None),
+                };
+                Ok(val)
+            }
+        },
+    );
+
+    registry.register_function_factory("get", |_, args_type| {
+        if args_type.len() != 2 {
+            return None;
+        }
```

Comment:
Argument constraint will be checked by type checker following the `signature.args_type`.
```suggestion
```